src/js/medium-editor.js

```diff
@@ -1220,6 +1220,11 @@ else if (typeof define === 'function' && define.amd) {
 
                     self.createLink(this, target, button);
                 }
+                else if (e.keyCode === 27) {
+                    e.preventDefault();
+                    self.showToolbarActions();
+                    restoreSelection.call(self, self.savedSelection);
```

Comment:
Yes, I'm suggesting something like that.  However, this is what is at the end of 'createLink', which is called when the user hits ENTER

```
this.checkSelection();
this.showToolbarActions();
input.value = '';
```

Ideally, `restoreEditorState` would be something that both `createLink` and your ESC handling would both call.